### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by icywind**
Migrated quickstart blog to Github guide

### CODE REVIEW

File: docs/GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+## Prerequisites:
+
+-   Apple VisionPro    
+-   Apple computer with Silicon CPU Chip (M1 or newer generation)
+-   Xcode (15.4)  with VisionOS Support
+-   Agora Developer account
+    
+## Setups
+
+1.  Clone the GitHub project from [this location](https://github.com/AgoraIO-Community/visionOS-Quickstart).
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: Since this guide is intended for a technical audience you should use more technical terms. Consider replacing with:
> Clone the project from this [GitHub. repo](https://github.com/AgoraIO-Community/visionOS-Quickstart).

File: docs/GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+2. Open the project with Xcode.
+3. Import the Agora SDK plugin. The project includes the Agora SDK as a dependency package. As soon as Xcode loads the project, the package download should start automatically and finish in a few minutes.
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: This step is a bit hard to follow. Consider adding a `:` instead of a period after the first statement and combing the following two sentences.

> Import the Agora SDK plugin: The project includes the Agora SDK as a dependency package and the package download should start automatically and finish in a few minutes.

File: docs/GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+![](/docs/images/blog_avp1_branch.png "package location")
+2.  CDN download     [https://download.agora.io/null/Agora_Native_SDK_for_iOS_rel.v4.2.6.133_39484_FULL_20240126_0204_291749.zip](https://download.agora.io/null/Agora_Native_SDK_for_iOS_rel.v4.2.6.133_39484_FULL_20240126_0204_291749.zip)
+    
+3.  Cocoa Pods
+```Makefile  
+	pod 'AgoraRtcEngine_Special_iOS', '4.2.6.133.VISION'
+```
+## Running the App
+
+Fill in your test mode Agora app ID in the appID field in VisionVideoCallView.swift.
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: To make the directions more clear, consider re-ordering the statements and separating out the "test mode" element. The "test mode" is pretty important, because if security is enabled the demo doesn't work. In the current sentence its easy for the user to miss.

To give some context, something like this:

> Open `VisionVideoCallView.swift.` and fill in your Agora app ID in the `appID` field. 

> > This guide is made for Agora projects without security enabled.

The note could use some better explanation but it at least makes it clear that they shouldn't have token security enabled.

File: docs/GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+![](/docs/images/blog_avp2_appid.png "appid entry")
+
+Build and run for either the Simulator or your physical VisionPro device. At the same time, prepare a second user to join the chat as a remote user.
+
+### Remote User Setup
+
+Use another device or use the Agora Web demo app for connecting to the video chat channel. The following screenshot shows a web user published video feed using the same App ID and channel name.
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: For the remote user, you should expand a bit more on these two options. Currently it's not really clear what app they should run on another device, is it this same demo or another demo? Also if they want to use the Agora Web Demo, there should be a link so the reader doesn't have to go searching for it.

File: docs/GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+```Makefile  
+	pod 'AgoraRtcEngine_Special_iOS', '4.2.6.133.VISION'
+```
+## Running the App
+
+Fill in your test mode Agora app ID in the appID field in VisionVideoCallView.swift.
+
+![](/docs/images/blog_avp2_appid.png "appid entry")
+
+Build and run for either the Simulator or your physical VisionPro device. At the same time, prepare a second user to join the chat as a remote user.
```
Comment (digitallysavvy) [inline docs/GUIDE.md:42]: Expand a bit more on the second sentence, explain to the reader why they need to prepare a second user. You mention the concept of "remote user" so you should explain what that means.

File: docs/GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+```xml
+<key>NSCameraUsageDescription</key>
+<string>camera for self capture</string>
+<key>NSMicrophoneUsageDescription</key>
+<string>mic for my voice</string>
+```
+  
+![](/docs/images/blog_avp5_plist.png "plist")
+
+You should confirm your permission to allow the app to use your VisionPro’s camera and microphone. Once you join the channel, you should see yourself in your avatar form and the remote user should show up on your screen!
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: You should break this first sentence down and expand upon it to give more clarity. For example when do the permissions get shown shown, and how do users join the channel? is it automatic or click a button?

File: docs/GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+![](/docs/images/blog_avp6_device.gif "device")
+
+The Web user can also see you in your digital form and you can start chatting together now!
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: This part is a bit unclear, you should provide some more context and be les specific to only Web.

In a previous instruction you gave the reader the option of using an app or web, but here you specifically only mention web. You should consider making this more generalized, ie. "On the other device/web browsers .." or "When the VisionPro user joins"

File: docs/GUIDE.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+![](/docs/images/blog_avp7_webuser.gif "web user")
+  
+
+## Basic Code Walkthrough
+
+Reading and understanding the code is never a boring task for developers! Here is a break-down of the project source so you can find the area of your interest quickly.
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: This intro sounds a bit robotic, consider revising to better explain to the user that this next section explains the code.

File: docs/GUIDE.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
+# VisionPro Meets the World: Video Chat with Agora's Video SDK
+
+ ![](/docs/images/blog_banner.webp "banner")
+
+The future is here, and you're holding it – the Apple VisionPro device. It's a marvel of technology, but sometimes, you just need to see a friendly face across the digital divide. That's where Agora's ubiquitous Real-Time Communication (RTC) SDK comes in. With Agora, video chat on your VisionPro device transcends physical boundaries. You can connect with other VisionPro users, and Agora's expansive reach allows you to chat seamlessly with people on entirely different platforms. Imagine video calling your friends and family on their smartphones, tablets, or laptops – all through the power of Agora on your VisionPro. This quick start guide will equip you to unlock the potential of Agora's native RTC SDK and transform your VisionPro into a powerful video chat hub.
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: Add a second paragraph that gives an overview of the structure of this guide. The guide starts with how to test the demo and then walks through explaining some of the code. You should let the user know that's the structure so they can better understand the path this guide is taking.

File: docs/GUIDE.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+
+|Class|Purpose|Notes|
+| --------   | -----:  | :----:  |
+Agora_VisionApp|Entry point of application|Standard entry point
+ContentView|UI View presenting to user at the beginning|Accepts the meeting’s channel name. Calls VisionVideoCallView
+VisionVideoCallView|Showing user’s video feeds as a grid format|Input the appID here
+AgoraVideoCanvasView|The video canvas view for rendering the video|This is a reusable class that works on iOS/MacOS as well.
+AgoraManager|The controller of the app business logic that implements Agora RTC Engine’s delegates.|Respond to join channel and leave channel; handles callback events like _didJoinedOfUid_ and _didJoinChannel_.
+
+You should see the code follows the logic flow described in the [official documentation](https://docs.agora.io/en/video-calling/get-started/get-started-sdk?platform=ios).
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: The logic is pretty important because it outlines how the demo is architected to work. I suggest bringing some of this information into the guide to give the user some of the basic understanding and then link out for more information.

File: docs/GUIDE.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+| --------   | -----:  | :----:  |
+Agora_VisionApp|Entry point of application|Standard entry point
+ContentView|UI View presenting to user at the beginning|Accepts the meeting’s channel name. Calls VisionVideoCallView
+VisionVideoCallView|Showing user’s video feeds as a grid format|Input the appID here
+AgoraVideoCanvasView|The video canvas view for rendering the video|This is a reusable class that works on iOS/MacOS as well.
+AgoraManager|The controller of the app business logic that implements Agora RTC Engine’s delegates.|Respond to join channel and leave channel; handles callback events like _didJoinedOfUid_ and _didJoinChannel_.
+
+You should see the code follows the logic flow described in the [official documentation](https://docs.agora.io/en/video-calling/get-started/get-started-sdk?platform=ios).  
+
+### Essential API Calls
```
Comment (digitallysavvy) [inline docs/GUIDE.md:140]: These are Agora specific and not really essential to the reader. This section should highlight key areas where the Agora SDK hooks into the AVP SDK and how that differs from traditional iOS/macOS implimentations.

File: docs/GUIDE.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+* Leave a channel
+```swift
+		self.agoraEngine.leaveChannel(leaveChannelBlock)
+```
+* Dispose of the engine
+```swift
+		AgoraRtcEngineKit.destroy()
+```
+
+### View Update Sequence
```
Comment (digitallysavvy) [inline docs/GUIDE.md]: This should go in the section above where you first reference the flow

File: docs/GUIDE.md [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+AgoraManager ->> AgoraManager : add to users
+RTCEngine ->> AgoraManager: didOfflineOfUid
+AgoraManager -x AgoraManager : remove from users
+AgoraManager ->> RTCEngine: LeaveChannel
+AgoraManager ->> RTCEngine: destroy
+AgoraManager -x AgoraManager : remove all users
+```
+
+
+## Conclusion
```
Comment (digitallysavvy) [inline docs/GUIDE.md:168]: The conclusion should offer a link out to either the Agora Docs or the next relevant piece of content the reader should visit to continue their journey.

**Review by digitallysavvy — CHANGES_REQUESTED** [review]
Overall great start, some minor notes to add more context so reader can better understand the what and why. The biggest change requested is about shifting the focusing of Essential API's section to focus more on the AVP specific elements of the implementation instead of the core Agora functions

File: docs/GUIDE.md (hunk 6 above)
Comment (icywind) [reply docs/GUIDE.md]: Added a new section on execution sequence.

**Review by digitallysavvy — APPROVED** [review]
Looks Good
